app/models/feed.rb

```diff
@@ -1,6 +1,8 @@
 class Feed < ApplicationRecord
   has_many :posts, -> { order(updated_at: :desc) },
     dependent: :destroy, inverse_of: :feed
+  has_one :last_post, -> { order(updated_at: :desc) },
+    class_name: "Post", dependent: nil, inverse_of: :feed
```

Comment:
since `posts` is already defined with an order (above), is this preferable to callers using `posts.first` or even

``` ruby
def domain
  posts.first.domain
end
```

?

Reply:
Yes, because the relationship macros created by ActiveRecord include automatic object caching if you call `last_post` twice, for example.

Reply:
having said that, we might actually need to reverse the sorting because has_one returns the first result?

Reply:
nope never mind we are putting the newest post first and has_one uses LIMIT 1 so this is right

Reply:
Also are we sure that this returns the most recent post as opposed to the oldest? Maybe worth a test to make sure domain changes over time are tracked properly.

Reply:
This seems to imply we are getting what we want, the most-recently-updated post:
```
irb(main):005:0> Feed.find_by(token: "rnbz7cl5oymjtzs7fs9cc5a1").posts.map &:updated_at
  Feed Load (11.4ms)  SELECT "feeds".* FROM "feeds" WHERE "feeds"."token" = $1 LIMIT $2  [["token", "[FILTERED]"], ["LIMIT", 1]]
  Post Load (9.0ms)  SELECT "posts".* FROM "posts" WHERE "posts"."feed_id" = $1 ORDER BY "posts"."updated_at" DESC  [["feed_id", 3]]   
=>                                                                                                                                     
[Thu, 13 Jan 2022 22:08:43.410627000 UTC +00:00,                                                                                       
 Wed, 12 Jan 2022 18:24:36.739165000 UTC +00:00,                                                                                       
 Tue, 11 Jan 2022 18:22:32.285917000 UTC +00:00,                                                                                       
 Tue, 11 Jan 2022 10:22:16.978643000 UTC +00:00]                                                                                       
irb(main):006:0> Feed.find_by(token: "rnbz7cl5oymjtzs7fs9cc5a1").last_post.created_at
  Feed Load (0.9ms)  SELECT "feeds".* FROM "feeds" WHERE "feeds"."token" = $1 LIMIT $2  [["token", "[FILTERED]"], ["LIMIT", 1]]
  Post Load (3.4ms)  SELECT "posts".* FROM "posts" WHERE "posts"."feed_id" = $1 ORDER BY "posts"."updated_at" DESC LIMIT $2  [["feed_id", 3], ["LIMIT", 1]]
=> Thu, 13 Jan 2022 22:08:43.410627000 UTC +00:00    
```